pkg/kubelet/metrics/collectors/volume_stats.go

```diff
@@ -120,7 +126,16 @@ func (collector *volumeStatsCollector) CollectWithStability(ch chan<- metrics.Me
 			addGauge(volumeStatsInodesDesc, pvcRef, float64(*volumeStat.Inodes))
 			addGauge(volumeStatsInodesFreeDesc, pvcRef, float64(*volumeStat.InodesFree))
 			addGauge(volumeStatsInodesUsedDesc, pvcRef, float64(*volumeStat.InodesUsed))
+			addGauge(volumeStatsHealthAbnormalDesc, pvcRef, convertBoolToFloat64(volumeStat.VolumeHealthStats.Abnormal), "abnormal")
```

Comment:
If the pvc is healthy I don't think we would want to report `abnormal` as a status.

I could suggest two approaches:
- remove the `volume_health_status` label in favor of renaming the metric to `kubelet_volume_stats_health_status_abnormal` and consider 1 as a value for PVC in an abnormal state and 0 for healthy PVC
- set the `volume_health_status` value depending on `VolumeHealthStats.Abnormal`. For example, for healthy PVC you could set it to `healthy` or something along those line

I would personally prefer the first approach.

Reply:
sorry, I think the current PR takes the second way. Right?

```
kubelet_volume_stats_health_status{namespace="testns",persistentvolumeclaim="testpvc",volume_health_status="abnormal"} 1

kubelet_volume_stats_health_status{namespace="testns",persistentvolumeclaim="testpvc",volume_health_status="abnormal"} 0
```

Reply:
Are you suggesting to take the first approach?

Reply:
cc @xing-yang

Reply:
oh, I see. I think I understand the #105585 (comment) wrongly. There are two approaches to add the metrics

1. use metric value 1 or 0 to indicate the healthy or unhealthy
2. use metrics value 1 or 0 to indicate something is true or not.

It seems like I mix both of them in my PR.

If my above understanding is correct, let me revise based on first approach that you suggested

Reply:
yes your understanding is correct

Reply:
Hi @dgrisonnet . Could you review the[ latest commit](https://github.com/kubernetes/kubernetes/pull/105585/commits/9808ae48a0a2aa46469f0262354dea127bf9746b)? I think I corrected them.